docs/design-documents/20221024-stream-inspector.md

```diff
@@ -0,0 +1,87 @@
+# Stream Inspector
+
+## Goals
+
+Make troubleshooting pipelines easier by making it possible to inspect the data which is flowing through pipeline.
+
+## Requirements
+
+1. All pipeline components (accessible "from the outside", i.e. sources, processors and destinations) should be
+   inspectable.
+    * For v1, it's enough that destinations are inspectable.
+2. A source inspector should show the records coming out of the respective source.
+3. A destination inspector should show the records coming into the respective destination.
+4. A processor has two inspectors: one for the incoming data and one for the resulting data (i.e. the transformed
+   records).
+5. The inspector should provide the data about the component being inspected, from the time the inspection started.
+   Historical data is not required.
```

Comment:
Is this what you mean? I want to make sure I understand this point.
```suggestion
5. The inspector should provide records flowing into/out of the component being inspected from when the inspection started.
   Historical data is not required.
```

Reply:
You're right. I'll then resolve this conversation, since we can continue in the comment you mention.:)